meta-intel-iot-security/meta-security-smack/recipes-core/systemd/systemd/0001-core-smack-add-support-for-defining-smack-rules-path-v229.patch

```diff
@@ -0,0 +1,211 @@
++
+ # ------------------------------------------------------------------------------
+ AC_ARG_ENABLE([gcrypt],
+         AS_HELP_STRING([--disable-gcrypt],[Disable optional GCRYPT support]),
+@@ -1607,6 +1622,7 @@ AC_MSG_RESULT([
+         Maximum System UID:      ${SYSTEM_UID_MAX}
+         Maximum System GID:      ${SYSTEM_GID_MAX}
+         Certificate root:        ${CERTIFICATEROOT}
++        Smack rules path:        ${DEFAULT_SMACK_RULES_PATH}
+ 
+         CFLAGS:                  ${OUR_CFLAGS} ${CFLAGS}
+         CPPFLAGS:                ${OUR_CPPFLAGS} ${CPPFLAGS}
+diff --git a/src/core/smack-setup.c b/src/core/smack-setup.c
+index 0c26e85..f1ad970 100644
+--- a/src/core/smack-setup.c
++++ b/src/core/smack-setup.c
+@@ -35,9 +35,22 @@
+ #include "smack-setup.h"
+ #include "string-util.h"
+ #include "util.h"
++#include "strv.h"
+ 
+ #ifdef HAVE_SMACK
+ 
++#ifndef STRINGIFY
++#  define STRINGIFY(foo) #foo
++#endif
++
++#define DEFAULT_RULES_PATH_SMACK STRINGIFY(DEFAULT_SMACK_RULES_PATH) "/smack/accesses.d/"
++#define DEFAULT_RULES_PATH_CIPSO STRINGIFY(DEFAULT_SMACK_RULES_PATH) "/smack/cipso.d/"
++#define DEFAULT_RULES_PATH_NETLABEL STRINGIFY(DEFAULT_SMACK_RULES_PATH) "/smack/netlabel.d/"
++
++#define RULES_PATH_SMACK "/etc/smack/accesses.d/"
++#define RULES_PATH_CIPSO "/etc/smack/cipso.d/"
++#define RULES_PATH_NETLABEL "/etc/smack/netlabel.d/"
++
+ static int write_access2_rules(const char* srcdir) {
+         _cleanup_close_ int load2_fd = -1, change_fd = -1;
+         _cleanup_closedir_ DIR *dir = NULL;
+@@ -133,6 +146,7 @@ static int write_cipso2_rules(const char* srcdir) {
+         int r = 0;
+ 
+         cipso2_fd = open("/sys/fs/smackfs/cipso2", O_RDWR|O_CLOEXEC|O_NONBLOCK|O_NOCTTY);
++
+         if (cipso2_fd < 0)  {
+                 if (errno != ENOENT)
+                         log_warning_errno(errno, "Failed to open '/sys/fs/smackfs/cipso2': %m");
+@@ -271,23 +285,35 @@ int mac_smack_setup(bool *loaded_policy) {
+ #ifdef HAVE_SMACK
+ 
+         int r;
++        _cleanup_strv_free_ char **smackpaths = NULL;
++        _cleanup_strv_free_ char **cipsopaths = NULL;
++        _cleanup_strv_free_ char **netlabelpaths = NULL;
++        char **smackpath;
++        char **cipsopath;
++        char **netlabelpath;
+ 
+         assert(loaded_policy);
+ 
+-        r = write_access2_rules("/etc/smack/accesses.d/");
+-        switch(r) {
+-        case -ENOENT:
+-                log_debug("Smack is not enabled in the kernel.");
+-                return 0;
+-        case ENOENT:
+-                log_debug("Smack access rules directory '/etc/smack/accesses.d/' not found");
+-                return 0;
+-        case 0:
+-                log_info("Successfully loaded Smack policies.");
+-                break;
+-        default:
+-                log_warning_errno(r, "Failed to load Smack access rules, ignoring: %m");
+-                return 0;
++        smackpaths = strv_new(DEFAULT_RULES_PATH_SMACK, RULES_PATH_SMACK, NULL);
```

Comment:
sure but someone can also put the rules twice to 2 different places.... but that check can be added.